enhancements/oc/mirroring-manifests.md

```diff
@@ -0,0 +1,260 @@
+When `--manifests-to-dir` is set, it takes precedence.
+When `--manifests-to-dir` is unset but `--to-dir` is set, the manifest directory is `${TO_DIR}/manifests`, as described in [the previous section](#pushing-to-disk).
+
+### User Stories
+
+#### Release-signature ConfigMaps
+
+Currently a cluster upgrade can be accomplished on a cluster that does not have an active connection to the internet.
+However manual steps are required to create a ConfigMap containing the signature data required for update image verification.
+This enhancement will automatically create the ConfigMap, so the user doesn't have to think about manual steps.
+
+##### Mirroring to a central registry in a fully air gapped environment
+
+A fully air gapped environment is one in which your cluster nodes cannot access the internet.
+For this reason you must mirror the images and manifests to a filesystem disconnected from that environment and then bring that host or removable media across that gap.
+Our documentation refers to this as disconnected mirroring.
+It may also be the case that there are multiple clusters within the air gapped network.
+In such a case it makes sense to configure a central registry in the air gapped environment from which every cluster can pull upgrade imagery.
+Assuming the aforementioned use case and the use of removable media that will be sneakernetted across the gap, the steps for mirroring are:
+
+1. Connect the removable media to a system connected to the internet.
+1. Mirror the images and manifests to a directory on the removable media, using [the `--to-dir` flow](#pushing-manifests-to-disk-for-later-application):
+    ```console
+    $ oc adm release mirror "--to-dir=${REMOVABLE_MEDIA_PATH}/mirror" quay.io/openshift-release-dev/ocp-release:4.3.0-x86_64
+    ```
+    The enhanced `mirror` command will output the path, similar to the example below, to a ConfigMap created containing the image signature.
+    ```console
+    Configmap <removable-media-path>/mirror/manifests/signature-sha256-81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92.yaml created
+    ```
+1. Run any required checks and scrubs on the removable media.
+1. Connect the removable media to a host within the air gapped environment that has access to both the central registry and to any cluster requiring upgrade.
+1. Upload the mirrored images to the central registry, as described in [the `--to-dir` flow](#pushing-manifests-to-disk-for-later-application):
+    ```console
+    $ oc image mirror "--from-dir=${REMOVABLE_MEDIA_PATH}/mirror" file://openshift/release:4.3.0* REGISTRY/REPOSITORY
+    ```
+1. Use `oc` to log in to a given cluster to be upgraded.
+1. Apply the the mirrored manifests to the connected cluster, including the above-mentioned signature ConfigMap, as described in [the `--to-dir` flow](#pushing-manifests-to-disk-for-later-application):
+    ```console
+    $ oc apply -Rf "${REMOVABLE_MEDIA_PATH}/mirror/manifests"
```

Comment:
As noted above I'd prefer to not have recursion for future extension purposes.

Reply:
> As noted above I'd prefer to not have recursion for future extension purposes.

Folding into [the earlier thread][1].

[1]: https://github.com/openshift/enhancements/pull/188#discussion_r393735144